app/Platform/Controllers/GameDevInterestController.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Controllers;
+
+use App\Community\Enums\UserGameListType;
+use App\Community\Models\UserGameListEntry;
+use App\Http\Controller;
+use App\Site\Enums\Permissions;
+use Illuminate\Contracts\View\View;
+use Illuminate\Http\Request;
+
+class GameDevInterestController extends Controller
+{
+    public function __invoke(Request $request): View
+    {
+        if (!$request->user()) {
+            abort(401);
+        }
+
+        $permissions = $request->user()->getAttribute('Permissions');
+        if ($permissions < Permissions::Developer) {
+            abort(401);
+        }
+
+        $gameId = (int) $request->route('game');
+        $gameData = getGameData($gameId);
+        if ($gameData === null) {
+            abort(404);
+        }
+
+        if ($permissions < Permissions::Moderator && !hasSetClaimed($request->user()->User, $gameId, true)) {
+            abort(401);
```

Comment:
nit: This should probably be a 403. The user is authenticated, but they are not authorized.